tests/integration_tests/test_botorch.py

```diff
@@ -78,9 +79,9 @@ def candidates_func(
         train_obj: torch.Tensor,
         train_con: Optional[torch.Tensor],
         bounds: torch.Tensor,
-    ) -> torch.Tensor:
+    ) -> Any:  # The return type is Any because we don't want mypy to check the type.
```

Comment:
So this fix is a kinda hotfix of mypy bug according to the issue description, right? Should we add a TODO comment?

Reply:
I'm not sure which bug of `mypy` we should watch for.

Reply:
hmm, how about [this](https://github.com/optuna/optuna/pull/3627#discussion_r8895076520)? I suppose the situation is similar to this case.